### ISSUE TITLE
Download of report on heroku not working anymore

### ISSUE THREAD

**Issue by designhunger** [masugadesign/lab-reports-craft-cms#8]
Hi - thanks a lot for the plugin, that helped us manage our data in the past really well. Of course we are using the pro version.

We just updated our CraftCMS to the newest version 4.5.5. We are running the app on heroku. Before it was possible to generate the reports and downloads them from the heroku storage. Of course data were deleted after a restart of the dynos.

But now the reports are not saved at all on the storage. Reports seems to be generated fine, as we do not see any errors in the logs - also with debug activated. We just see the message "Unavailable (File Missing)" in the reports details of your plugin.

Is it possible to somehow activate AWS S3 storage? Has anything changed in the plugin or is there any configuration option besides the ones mentioned in the documentation?

Appreciate your help - thanks!

**Comment by benjaminkohl** [comment]
Nothing has changed with how the plugin stores generated reports. It is based on the `fileStorageFolder` plugin setting which defaults to the "labreports" folder in the Craft storage path. Maybe the plugin is failing to generate that subfolder? If so, the labreports.log file in the storage/logs folder should contain a log line that looks like _"Error creating Lab Reports folder at: `PATH`"_.

At this time, it isn't possible to set an AWS S3 folder as the report storage folder.

**Comment by designhunger** [comment]
Thanks for the fast response @benjaminkohl ! 

We just found out that it's not working due to another change. We switched the queue to a background worker on heroku. There is no possibility to get access to the filesystem of the worker, so all generated reports can not be accessed.

So, I close the issue here. But this also means, that we will need to switch to another plugin to generated reports :(
